src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -225,6 +227,8 @@ public void initialize() {
         if (options.SHARED_OBJECTS_ENABLED && options.SHARED_OBJECTS_FORCE) {
             sharedObjects.startSharing();
         }
+
+        initialized = true;
```

Comment:
The CoreLibrary already manages a loading state, this could be added to it.
But this is probably simpler, so it's fine too.

Reply:
It also applies after the core library is loaded though, and we're in that post-boot phase.

Reply:
Yes, it would need to be a new state.